muzzle/src/main/java/io/opentelemetry/javaagent/tooling/ByteArrayUrl.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.net.URLStreamHandler;
+import java.nio.charset.StandardCharsets;
+import java.security.Permission;
+import java.security.PrivilegedAction;
+
+public class ByteArrayUrl {
+
+  private ByteArrayUrl() {}
+
+  private static final String URL_SCHEMA = "x-otel-binary";
+
+  @SuppressWarnings("removal")
+  public static URL create(String uniqueName, byte[] data) {
```

Comment:
I don't get why these would need to have a unique name

Reply:
This was just done based on what ByteBuddy does. I assume it is just a nice-to-have so that `toString()` returns something which allows users to identify the content when debugging.

Reply:
Let me rephrase. Why is the parameter called `uniqueName`? Does it really have to be unique?

Reply:
Ah I see. No it doesn't have to be unique. I'll change it to `contentName`.